lib/readline.js

```diff
@@ -134,6 +136,8 @@ function Interface(input, output, completer, terminal) {
     }
     crlfDelay = input.crlfDelay;
     input = input.input;
+  } else {
+    throw new ERR_MISSING_OPTION('options.input');
```

Comment:
This is not related to the other changes to this file, is it? Is there any particular reason for letting go of the legacy interface?

If we do this, we should probably also rename `input` to `options`, and get rid of the then-unused function parameters.

Reply:
This checks for `input.input` field to be defined since its a required field. The input parameter here is actually indicated as `options` in the docs, therefore it made sense to go with `options.input`.

The docs state that this method only takes in one `options` object which has fields (such as input, output, completer, terminal) which are at the moment passed in separately as you can see above.

I think we should go ahead with renaming `input` to `options`, and get rid of the then-unused function parameters.

Reply:
@rexagod Okay, but like I said – It’s an opinionated breaking change, so t shouldn’t be mixed in with a bugfix. That the legacy signature is undocumented isn’t a good reason to remove it, imo.